errbot/backends/slack.py

```diff
@@ -650,16 +651,16 @@ def build_identifier(self, txtrep):
         log.debug("building an identifier from %s" % txtrep)
         username, userid, channelname, channelid = self.extract_identifiers_from_string(txtrep)
 
+        if userid is None and username is not None:
+            userid = self.username_to_userid(username)
+        if channelid is None and channelname is not None:
+            channelid = self.channelname_to_channelid(channelname)
+        if userid is not None and channelid is not None:
+            return SlackRoomOccupant(self.sc, userid, channelid, bot=self)
```

Comment:
That was my initial thought as well, but it caused a bunch of errors further down the line when I tried it. I will give it another try to see if it would be reasonable.